Modules/System/Azure Queues/src/AzureStorageQueuesImpl.Codeunit.al

```diff
@@ -0,0 +1,293 @@
+        if HttpResponseMessage.IsSuccessStatusCode then begin
+            if HttpResponseMessage.Content.ReadAs(ResponseText) then
+                exit(GetQueuesListFromResponseBody(ResponseText));
+        end;
+    end;
+
+    procedure DeleteQueue(StorageAccountName: Text; Queue: Text): Boolean
+    var
+        HttpClient: HttpClient;
+        HttpResponseMessage: HttpResponseMessage;
+        HttpContent: HttpContent;
+    begin
+        HttpClient.Delete(GetQueueEndpoint(StorageAccountName, Queue), HttpResponseMessage);
+        exit(HttpResponseMessage.IsSuccessStatusCode);
+    end;
+
+    procedure CheckIfQueueExists(StorageAccountName: Text; Queue: Text): Boolean
+    var
+        HttpClient: HttpClient;
+        HttpResponseMessage: HttpResponseMessage;
+        HttpContent: HttpContent;
+    begin
+        HttpClient.Put(GetQueueEndpoint(StorageAccountName, Queue), HttpContent, HttpResponseMessage);
+        exit(HttpResponseMessage.IsSuccessStatusCode);
+    end;
+
+    procedure PostMessageToQueue(StorageAccountName: Text; Queue: Text; MessageBody: Text): Boolean
+    var
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+        HttpContent: HttpContent;
+        HttpRequest: HttpRequestMessage;
+        HttpResponse: HttpResponseMessage;
+        ResponseText: Text;
+    begin
+        if CheckIfQueueExists(StorageAccountName, Queue) then begin
+            HttpContent.WriteFrom('<QueueMessage><MessageText>' + MessageBody + '</MessageText></QueueMessage>');
+            HttpClient.Post(GetAzureQueueMessagesEndpoint(StorageAccountName, Queue), HttpContent, HttpResponse);
+            exit(HttpResponse.IsSuccessStatusCode);
+        end;
+    end;
+
+    procedure UpdateMessageToQueue(StorageAccountName: Text; Queue: Text; MessageId: Text; PopReceipt: Text; NewMessageBody: Text): Boolean
+    var
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+        HttpContent: HttpContent;
+        HttpRequest: HttpRequestMessage;
+        HttpResponse: HttpResponseMessage;
+        ResponseText: Text;
+    begin
+        if CheckIfQueueExists(StorageAccountName, Queue) then begin
+            HttpContent.WriteFrom('<QueueMessage><MessageText>' + NewMessageBody + '</MessageText></QueueMessage>');
+            HttpClient.Put(GetAzureQueueUpdateMessageEndpoint(StorageAccountName, Queue, MessageId, PopReceipt), HttpContent, HttpResponse);
+            exit(HttpResponse.IsSuccessStatusCode);
+        end;
+    end;
+
+    procedure GetNextMessageFromQueue(StorageAccountName: Text; Queue: Text[20]): Text
+    var
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+        HttpContent: HttpContent;
+        HttpRequest: HttpRequestMessage;
+        HttpResponse: HttpResponseMessage;
+        ResponseText: Text;
+    begin
+        if CheckIfQueueExists(StorageAccountName, Queue) then begin
```

Comment:
An if-not-then-exit pattern would be neater here, I think. This goes for many other procedures in this codeunit and would bring it more in line with the AL Guidelines.